src/dialogs/config-flow/dialog-config-flow.ts

```diff
@@ -0,0 +1,377 @@
+    let bodyContent: TemplateResult | undefined;
+    let buttonContent: TemplateResult | undefined;
+    let descriptionKey: string | undefined;
+
+    if (!step) {
+      bodyContent = html`
+        <div class="init-spinner">
+          <paper-spinner active></paper-spinner>
+        </div>
+      `;
+    } else if (step.type === "abort") {
+      descriptionKey = `component.${step.handler}.config.abort.${step.reason}`;
+      headerContent = "Aborted";
+      bodyContent = html``;
+      buttonContent = html`
+        <mwc-button on-click="_flowDone">Close</mwc-button>
+      `;
+    } else if (step.type === "create_entry") {
+      descriptionKey = `component.${
+        step.handler
+      }.config.create_entry.${step.description || "default"}`;
+      headerContent = "Success!";
+      bodyContent = html`
+        <p>Created config for ${step.title}</p>
+      `;
+      buttonContent = html`
+        <mwc-button on-click="_flowDone">Close</mwc-button>
+      `;
+    } else {
+      // form
+      descriptionKey = `component.${step.handler}.config.step.${
+        step.step_id
+      }.description`;
+      headerContent = localize(
+        `component.${step.handler}.config.step.${step.step_id}.title`
+      );
+      bodyContent = html`
+        <ha-form
+          .data=${this._stepData}
+          @data-changed=${this._stepDataChanged}
+          .schema=${step.data_schema}
+          .error=${step.errors}
+          .computeLabel=${this._labelCallback}
+          .computeError=${this._errorCallback}
+        ></ha-form>
+      `;
+
+      const allRequiredInfoFilledIn =
+        this._stepData &&
+        step.data_schema.every(
+          (field) =>
+            field.optional ||
+            !["", undefined].includes(this._stepData![field.name])
+        );
+
+      buttonContent = this._loading
+        ? html`
+            <div class="submit-spinner">
+              <paper-spinner active></paper-spinner>
+            </div>
+          `
+        : html`
+            <div>
+              <mwc-button
+                @click=${this._submitStep}
+                .disabled=${!allRequiredInfoFilledIn}
+              >
+                Submit
+              </mwc-button>
+
+              ${!allRequiredInfoFilledIn
+                ? html`
+                    <paper-tooltip position="left">
+                      Not all required fields are filled in.
+                    </paper-tooltip>
+                  `
+                : html``}
+            </div>
+          `;
+    }
+
+    let description: string | undefined;
+
+    if (step && descriptionKey) {
+      const args: [string, ...string[]] = [descriptionKey];
+      const placeholders = step.description_placeholders || {};
+      Object.keys(placeholders).forEach((key) => {
+        args.push(key);
+        args.push(placeholders[key]);
+      });
+      description = localize(...args);
+    }
+
+    return html`
+      <paper-dialog
+        with-backdrop
+        .opened=${true}
```

Comment:
always true?

Reply:
Yes, because people can click outside the dialog and that will set opened to false. We want the dialog to show as we will remove all DOM elements when there is nothing to show.